File: backend/src/main/java/com/springbootjwtpostgres/backend/bill/BillSearchCriteria.java
package com.springbootjwtpostgres.backend.bill;

import com.springbootjwtpostgres.backend.order.Order;
import com.springbootjwtpostgres.backend.order.OrderStatus;
import com.springbootjwtpostgres.backend.order.OrderType;
import lombok.Data;

import java.util.Date;

@Data
public class BillSearchCriteria {
    private OrderType billType;
    private OrderStatus billStatus;
    private double billTotalPriceFrom;
    private double billTotalPriceTo;
    private Date createdAtFrom;
    private Date createdAtTo;
    private Order order;
}

File: frontend/src/app/models/bill/BillSearchCriteria.ts
import { EOrderType } from '../order/EOrderType';
import { EOrderStatus } from '../order/EOrderStatus';

export class BillSearchCriteria {
  billType?: EOrderType;
  billStatus?: EOrderStatus;
  billTotalPriceFrom?: number;
  billTotalPriceTo?: number;
  createdAtFrom?: Date;
  createdAtTo?: Date;
  orderId?: number;
}
